### ISSUE TITLE
The pom produced by MP 4.0 is not useful any more

### ISSUE THREAD

**Issue by Emily-Jiang** [eclipse/microprofile#205]
Due to the fix of provided scope https://github.com/eclipse/microprofile/pull/183 as part of the [issue](https://github.com/eclipse/microprofile/issues/158), this leads to the pom produced by MP 4.0 is not useful any more.
```
<!-- https://mvnrepository.com/artifact/org.eclipse.microprofile/microprofile -->
<dependency>
    <groupId>org.eclipse.microprofile</groupId>
    <artifactId>microprofile</artifactId>
    <version>4.0</version>
    <type>pom</type>
    <scope>provided</scope>
</dependency>

```
The above dependency does not pull in any dependencies, so the previous working applications don't work any more.

```
<!-- https://mvnrepository.com/artifact/org.eclipse.microprofile/microprofile -->
<dependency>
    <groupId>org.eclipse.microprofile</groupId>
    <artifactId>microprofile</artifactId>
    <version>3.3</version>
    <type>pom</type>
    <scope>provided</scope>
</dependency>
```
We need to fix this as soon as possible.

**Comment by ederks85** [comment]
PR https://github.com/eclipse/microprofile/pull/206

**Comment by ederks85** [comment]
MP 4.0.1-SNAPSHOT
```
<dependency>
      <groupId>org.eclipse.microprofile.*</groupId>
      <artifactId>microprofile-*-api</artifactId>
</dependency>
```

```
[INFO] org.eclipse.microprofile:microprofile-spec:pom:4.0.1-SNAPSHOT
[INFO] +- jakarta.enterprise:jakarta.enterprise.cdi-api:jar:2.0.2:compile
[INFO] |  +- jakarta.interceptor:jakarta.interceptor-api:jar:1.2.5:compile
[INFO] |  \- jakarta.inject:jakarta.inject-api:jar:1.0:compile
[INFO] +- jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile
[INFO] +- jakarta.json.bind:jakarta.json.bind-api:jar:1.0.2:compile
[INFO] +- jakarta.json:jakarta.json-api:jar:1.1.6:compile
[INFO] +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
[INFO] +- org.eclipse.microprofile.config:microprofile-config-api:jar:2.0:compile
[INFO] +- org.eclipse.microprofile.fault-tolerance:microprofile-fault-tolerance-api:jar:3.0:compile
[INFO] +- org.eclipse.microprofile.health:microprofile-health-api:jar:3.0:compile
[INFO] +- org.eclipse.microprofile.metrics:microprofile-metrics-api:jar:3.0:compile
[INFO] +- org.eclipse.microprofile.jwt:microprofile-jwt-auth-api:jar:1.2:compile
[INFO] +- org.eclipse.microprofile.openapi:microprofile-openapi-api:jar:2.0:compile
[INFO] +- org.eclipse.microprofile.rest.client:microprofile-rest-client-api:jar:2.0:compile
[INFO] \- org.eclipse.microprofile.opentracing:microprofile-opentracing-api:jar:2.0:compile
```

demo-service

```
    <dependency>
      <groupId>org.eclipse.microprofile</groupId>
      <artifactId>microprofile</artifactId>
      <version>4.0.1-SNAPSHOT</version>
      <type>pom</type>
      <scope>provided</scope>
    </dependency>
```
	
```
[INFO] com.example:demo-service:war:1.0-SNAPSHOT
[INFO] \- org.eclipse.microprofile:microprofile:pom:4.0.1-SNAPSHOT:provided
[INFO]    +- jakarta.enterprise:jakarta.enterprise.cdi-api:jar:2.0.2:provided
[INFO]    |  +- jakarta.interceptor:jakarta.interceptor-api:jar:1.2.5:provided
[INFO]    |  \- jakarta.inject:jakarta.inject-api:jar:1.0:provided
[INFO]    +- jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:provided
[INFO]    +- jakarta.json.bind:jakarta.json.bind-api:jar:1.0.2:provided
[INFO]    +- jakarta.json:jakarta.json-api:jar:1.1.6:provided
[INFO]    +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:provided
[INFO]    +- org.eclipse.microprofile.config:microprofile-config-api:jar:2.0:provided
[INFO]    +- org.eclipse.microprofile.fault-tolerance:microprofile-fault-tolerance-api:jar:3.0:provided
[INFO]    +- org.eclipse.microprofile.health:microprofile-health-api:jar:3.0:provided
[INFO]    +- org.eclipse.microprofile.metrics:microprofile-metrics-api:jar:3.0:provided
[INFO]    +- org.eclipse.microprofile.jwt:microprofile-jwt-auth-api:jar:1.2:provided
[INFO]    +- org.eclipse.microprofile.openapi:microprofile-openapi-api:jar:2.0:provided
[INFO]    +- org.eclipse.microprofile.rest.client:microprofile-rest-client-api:jar:2.0:provided
[INFO]    \- org.eclipse.microprofile.opentracing:microprofile-opentracing-api:jar:2.0:provided
```

**Comment by ederks85** [comment]
This way, the project using the single MP dependency will have access to every dependency declared there. By defining the scope as "provided" in the dependency defining project, every MP dependency will come in as "Provided". Therefore, no MP API artifacts will end up in the final build artifacts.

**Comment by Emily-Jiang** [comment]
Closed this issue as the fix was made to the [release 4.0.1](https://github.com/eclipse/microprofile/releases/tag/4.0.1).
